Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -164,19 +185,47 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
             log.error("Failed to perform fetch request with error: \(error). This is an internal error.")
         }
     }
+
+    private func channelBelongsToController(_ channel: ChatChannel, change: ListChange<ChatChannel>) -> Bool {
+        if let filter = filter {
+            return filter(channel)
+        }
+
+        // Given that at the moment some delegate methods are not yet removed, but some integrators are still using
+        // those, we need to keep using them for now.
+        // This block should be removed once `shouldAddNewChannelToList` and `shouldListUpdatedChannel` methods are
+        // fully removed.
+        let deprecatedFallback: () -> Bool? = { [unowned self] in
+            switch change {
+            case .insert:
+                return self.delegate?.controller(self, shouldAddNewChannelToList: channel)
+            case .update:
+                return self.delegate?.controller(self, shouldListUpdatedChannel: channel)
+            default:
+                return nil
+            }
+        }
+
+        do {
+            return try deprecatedFallback() ?? channel.meets(query.filter)
+        } catch {
+            log.error("Unable to resolve complex filter. Please pass a `filter` block when intializing ChatChannelListController")
+        }
+        return true
```

Comment:
nit: `meets` is empty and most likely we won't implement it. If we make it non-throwing the code would get a bit simpler:
```suggestion
        return deprecatedFallback() ?? channel.meets(query.filter)
```

Reply:
The idea is for us to provide a simple non expensive implementation for it in the future, and throw in the case of it getting to complex to resolve in a reasonable time